drivers/storage/portworx/portworx.go

```diff
@@ -181,6 +181,15 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1alpha1.StorageClu
 		}
 	}
 
+	// If prometheus is enabled and metrics are not enabled explicitly, then
+	// enabled exporting of metrics
+	if toUpdate.Spec.Monitoring != nil &&
```

Comment:
What happens when Spec.Monitoring.EnableMetrics is set to true, but Prometheus is nil ?